### PR TITLE
Imageset subscription config support

### PR DESCRIPTION
**Pull request by ashishmax31**
Allow addon imageset's to define subscription configs

### CODE REVIEW

**Comment by yashvardhan-kukreja** [comment]
Awesome work! @ashishmax31. 

Just a minor nit: can you run `make generate` to auto-generate some required changes complying with your PR :)

**Review by yashvardhan-kukreja — APPROVED** [review]

**Comment by sugarraysam** [comment]
> Awesome work! @ashishmax31.
> 
> Just a minor nit: can you run `make generate` to auto-generate some required changes complying with your PR :)

@ashish you probably want to install pre-commit hooks:
```bash
$ pre-commit install
```

File: managedtenants/core/addons_loader/addon.py [hunk 1]
```diff
@@ -66,7 +66,23 @@ def name(self):
...
-        return self.metadata.get("config") is not None
+        # If imageset is present, check for subscriptionConfig in the
+        # imageset file, otherwise check for the default subscription config
+        # in the addon metadata file.
+        if self.imageset:
+            if self.imageset.get("subscriptionConfig"):
+                return True
+        return self.metadata.get("subscriptionConfig") is not None
+
+    def get_subscription_config(self):
```
Comment (sugarraysam) [inline managedtenants/core/addons_loader/addon.py:77]: nit: do we need the `subscription_config_present` function? it seems in any case, the `get_subscription_config` will return `None` if the config is not there, so we could really just do:

```python
def get_subscription_config(self):
  if self.imageset and self.imageset.get("subscriptionConfig"):
      return self.imageset.get("subscriptionConfig", None)
  return self.metadata.get("subscriptionConfig", None)
```

**Review by sugarraysam — APPROVED** [review]
lgtm, good work !

File: managedtenants/core/addons_loader/addon.py (hunk 1 above)
Comment (ashishmax31) [reply managedtenants/core/addons_loader/addon.py:68]: Yep, that works

**Pull request merged**